pkg/scheduler/objects/queue_test.go

```diff
@@ -1338,10 +1340,203 @@ func TestSupportTaskGroup(t *testing.T) {
 	assert.Assert(t, !leaf.SupportTaskGroup(), "leaf queue (FAIR policy) should not support task group")
 }
 
-func TestGetPartitionQueues(t *testing.T) {
+func TestGetPartitionQueueDAOInfo(t *testing.T) {
 	root, err := createRootQueue(nil)
 	assert.NilError(t, err, "failed to create basic root queue: %v", err)
-	root.properties = make(map[string]string)
-	root.properties["key"] = "value"
-	assert.Equal(t, "value", root.GetPartitionQueues().Properties["key"])
+
+	// test properties
+	root.properties = getProperties()
+	assert.Assert(t, reflect.DeepEqual(root.properties, root.GetPartitionQueueDAOInfo().Properties))
+
+	// test template
+	root.template = template.NewTemplate(getProperties(), getResource(t), getResource(t))
+	assert.Assert(t, reflect.DeepEqual(root.template.GetProperties(), root.GetPartitionQueueDAOInfo().TemplateInfo.Properties))
+	assert.Equal(t, root.template.GetMaxResource().DAOString(), root.GetPartitionQueueDAOInfo().TemplateInfo.MaxResource)
```

Comment:
This is dangerous: resources are a map you cannot never rely on the fact that the resources will be returned in a fixed order. Two calls to DAOString() on the same resource might return different string values.

Reply:
you are right.

There is a similar issue in code base (https://github.com/apache/incubator-yunikorn-core/blob/master/pkg/common/resources/resources_test.go#L1550). Maybe we should find out if it contains the expected string one by one. Or we can make DAOString() have consistent order. The benefit is the restful response get more readable. WDTY? I will file a PR to fix existent code.

Reply:
We need a follow up PR. I would suggest that calls that do not directly test `DAOString()` we use single resource values.
In the test case for for `DAOString()` only test the nil, empty and single value cases for full equality and a multi valued resource for the layout markers. This might need to be done for more tests.

Reply:
I created a ticket for existent code (https://issues.apache.org/jira/browse/YUNIKORN-797)

in this PR, we test `DAOString()` only if it uses single value.